pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1351,6 +1352,11 @@ async def parse_advisories(self, result):
 
                 for collection in update.collections:
                     coll_dict = UpdateCollection.createrepo_to_dict(collection)
+                    if coll_dict["name"] is None:
+                        coll_dict["name"] = (
+                            "collection-autofill-"
+                            + hashlib.sha256(update_record.fromstr.encode("utf-8")).hexdigest()[:12]
```

Comment:
Same here. Also, I was thinking "we can just use the pulp_id as a unique-ifier" - but at this point in sync, I don't believe we've persisted the object yet, I don't know that it *has* a pulp_id yet? If it does, then using the object's pulp_id as the Unique Thing is the way to go.

Reply:
It is possible use the `pulp_id` from the `UpdateRecord` which should be unique for every collection then doesn't it?

Reply:
In that case one could just provide the prefix + `pulp_id`, or prefix + a hash of the `pulp_id`.

Reply:
Just to record this from our discussion in matrix, for Future Engineers looking at this:
"you can't rely on the UpdateRecord - it's the advisory, and a single advisory can have multiple UpdateCollections, so you can't rely on id/digest/pulp_id from the UpdateRecord to create a unique-name for a single UpdateCollection"